### PR TITLE
Add docs on ResourceManagerHelper

### PR DESCRIPTION
**Pull request by Seppe-Vanhee**

### CODE REVIEW

**Review by VanRobaysWouter — APPROVED** [review]

File: user-guide/Standard_Apps/SRM/Interacting_with_SRM_using_code/srm_using_resourcemanagerhelper.md [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+|NotAllowed| The user does not have the necessary permissions to create, read, update or delete the ``ReservationInstance``. | *SubjectId* contains the ID of the ``ReservationInstance``.<br>*Message* will contain more details about the missing permission. |
+|ServiceDefinitionTypeDoesNotMatch|The ``ServiceDefinitionType`` of the ``ReservationInstance`` does not match with the type defined on the service definition. |*ReservationInstanceType* contains the type on the ``ReservationInstance``. |
+
+### Errors when adding or updating Resources
+
+| Error reason   | Description  | Available properties |
+|---|---|---|
+|  NotLicensed | The necessary license(s) are missing. To create a ``Resource`` the *ResourceManager* license is required, to create a ``FunctionResource`` the *ServiceManager* license is required as well. | None |
+| InitializeFunctionResourceFailed | Occurs when a FunctionResource could not be initialized. | *SubjectId* contains the ID of the ``Resource`` that failed to initialize.<br>*InitializeFunctionResourceResult* contains a more detailed reason, see the [table with possible values](#possible-values-for-the-initializefunctionresourceresult-property). |
```
Comment (slc-peter-vanpoucke) [inline user-guide/Standard_Apps/SRM/Interacting_with_SRM_using_code/srm_using_resourcemanagerhelper.md:347]: S: ``FunctionResource``

File: user-guide/Standard_Apps/SRM/Interacting_with_SRM_using_code/srm_using_resourcemanagerhelper.md [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+}
+```
+
+## Updating ReservationInstance properties
+
+Next to methods to perform the regular add, update and delete operations on ``Resources`` and ``ReservationInstances``, the helper also provides a method to only update the properties of a ``ReservationInstance``. This method will validate if no properties would unintentionally be overwritten, by comparing the old properties passed to the method with the existing properties on the booking. The ResourceManager will also perform less validation compared to a regular booking update, making this call more performant.
+
+```csharp
+var bookingId = Guid.Parse("...");
+var oldProperties = reservationInstance.Properties; // The current properties on the booking we intent to overwrite
```
Comment (Skyline-ThomasGH) [inline user-guide/Standard_Apps/SRM/Interacting_with_SRM_using_code/srm_using_resourcemanagerhelper.md:281]:
```suggestion
var oldProperties = reservationInstance.Properties; // The current properties on the booking we intend to overwrite
```

**Review by slc-peter-vanpoucke — CHANGES_REQUESTED** [review]

**Review by Skyline-ThomasGH — CHANGES_REQUESTED** [review]
Tested the code blocks which revealed some small errors. I had to include some namespaces, that to me were obvious, but there are sometimes two namespaces that contain the same class (the correct and the proto generator one). Not sure if we should clarify this more.

File: user-guide/Standard_Apps/SRM/Interacting_with_SRM_using_code/srm_using_resourcemanagerhelper.md [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+The ``ResourceManagerHelper`` provides a ``GetEligibleResources`` method to retrieve the ``Resources`` with certain capacities or capabilities that are available at a certain time. The ``EligibleResourceContext`` passed to this method contains a number of properties that can be used to fine-tune the filtering.
+
+The most basic filter is to specify the time range in which the ``Resources`` need to be available:
+
+```csharp
+// The timings in the context should be in UTC
+var now = DateTime.UtcNow;
+var context = new EligibleResourceContext()
+{
+    TimeRange = new TimeRangeUtc(now.AddHours(1), now.AddHours(4)),
```
[the remark this answers is not in the document]
Comment (Seppe-Vanhee) [reply user-guide/Standard_Apps/SRM/Interacting_with_SRM_using_code/srm_using_resourcemanagerhelper.md:156]: Added a comment in each code snippet if a specific namespace is needed. I think it's easier to clarify which namespaces are required for which code that way.

**Review by Skyline-ThomasGH — APPROVED** [review]

File: user-guide/Standard_Apps/SRM/Interacting_with_SRM_using_code/srm_using_resourcemanagerhelper.md [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+var resourceFilter = ResourceExposers.Name.Contains("Encoder").FunctionResourceExposers.FunctionGUID.Equal(Guid.Parse("..."));
+
+// Filter that matches all ReservationInstances that have overlap with the timerange from now until 3 hours from now
+var start = DateTime.UtcNow;
+var end = start.AddHours(3);
+var timeRangeFilter = ReservationInstanceExposers.Start.LessThan(end).AND(ReservationInstanceExposers.End.GreaterThan(start));
+```
+
+> [!IMPORTANT]
+> On a DataMiner Agent that is using OpenSearch or Elasticsearch, there is a default limit of 1024 clauses in a query. This means that you can only concatenate a maximum of 1024 filters using an "OR" filter. If this limit is not sufficient, you can adjust it using the "indices.query.bool.max_clause_count" option in [OpenSearch](https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/index-settings/) or [Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/search-settings.html).
```
[the remark this answers is not in the document]
Comment (slc-peter-vanpoucke) [reply user-guide/Standard_Apps/SRM/Interacting_with_SRM_using_code/srm_using_resourcemanagerhelper.md:53]: @Skyline-ThomasGH what is your opinion?

**Review by slc-peter-vanpoucke — CHANGES_REQUESTED** [review]

**Comment by slc-peter-vanpoucke** [comment]
ok

**Review by slc-peter-vanpoucke — APPROVED** [review]

**Review by MariekeGO — APPROVED** [review]

**Pull request merged**